<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AddObjectWindow</class>
 <widget class="QMainWindow" name="AddObjectWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabShape">
       <enum>QTabWidget::Triangular</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Sphere</string>
       </attribute>
       <layout class="QFormLayout" name="formLayout">
        <item row="1" column="0">
         <widget class="QLabel" name="labelPosition">
          <property name="toolTip">
           <string>Position in 3D space of the center of the sphere</string>
          </property>
          <property name="text">
           <string>Position </string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLineEdit" name="linePositionX">
            <property name="text">
             <string/>
            </property>
            <property name="placeholderText">
             <string>Enter x coordinate</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="linePositionY">
            <property name="text">
             <string/>
            </property>
            <property name="placeholderText">
             <string>Enter y coordinate</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="linePositionZ">
            <property name="text">
             <string/>
            </property>
            <property name="placeholderText">
             <string>Enter z coordinate</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="labelMaterial">
          <property name="toolTip">
           <string>Material that the sphere will be made of</string>
          </property>
          <property name="text">
           <string>Material</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QComboBox" name="comboBox">
          <item>
           <property name="text">
            <string>Glass</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Diffuse material</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Metal</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Light source</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QGroupBox" name="groupGlass">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Glass properties</string>
          </property>
          <layout class="QFormLayout" name="formLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="labelIoR_2">
             <property name="toolTip">
              <string>Index of refraction (at least 1.0)</string>
             </property>
             <property name="text">
              <string>Index of refraction</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="lineIoR">
             <property name="text">
              <string/>
             </property>
             <property name="placeholderText">
              <string>Index of refraction of the glass (at least 1.0)</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="lineReflectivity">
             <property name="placeholderText">
              <string>Reflectivity of the glass (between 0.0 and 1.0)</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLineEdit" name="lineAbsorptionR">
               <property name="placeholderText">
                <string>Absorption of red color (at least 0.0)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineAbsorptionG">
               <property name="placeholderText">
                <string>Absorption of green color (at least 0.0)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineAbsorptionB">
               <property name="placeholderText">
                <string>Absorption of blue color (at least 0.0)</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="labelReflectivity">
             <property name="toolTip">
              <string>How much the glass reflects (betwen 0.0 and 1.0)</string>
             </property>
             <property name="text">
              <string>Reflectivity</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="labelAbsorption">
             <property name="toolTip">
              <string>How much of each color channel the glass absorb</string>
             </property>
             <property name="text">
              <string>Absorption</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QGroupBox" name="groupMetal">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Metal properties</string>
          </property>
          <layout class="QFormLayout" name="formLayout_3">
           <item row="0" column="0">
            <widget class="QLabel" name="label_7">
             <property name="toolTip">
              <string>Color of the metal it were not reflecting part of the light</string>
             </property>
             <property name="text">
              <string>Albedo</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_8">
             <property name="toolTip">
              <string>To what extent the metal doesn't reflect like a mirror, how shady it is</string>
             </property>
             <property name="text">
              <string>Fuzziness</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="lineFuzziness">
             <property name="placeholderText">
              <string>How much it reflects not like a mirror (between 0.0 and 1.0)</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLineEdit" name="lineMetalAlbedoR">
               <property name="placeholderText">
                <string>Red part of the metal's color (&gt;0.0)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineMetalAlbedoG">
               <property name="placeholderText">
                <string>Green part of the metal's color (&gt;0.0)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineMetalAlbedoB">
               <property name="placeholderText">
                <string>Blue part of the metal's color (&gt;0.0)</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QGroupBox" name="groupDiffuse">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Diffuse material properties</string>
          </property>
          <layout class="QFormLayout" name="formLayout_4">
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="toolTip">
              <string>Color of the diffuse material</string>
             </property>
             <property name="text">
              <string>Albedo</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLineEdit" name="lineDiffuseR">
               <property name="placeholderText">
                <string>Red part of the color (&gt;0.0)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineDiffuseG">
               <property name="placeholderText">
                <string>Green part of the color (&gt;0.0)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineDiffuseB">
               <property name="placeholderText">
                <string>Blue part of the color (&gt;0.0)</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="7" column="1">
         <widget class="QGroupBox" name="groupLightSource">
          <property name="title">
           <string>Light source properties</string>
          </property>
          <layout class="QFormLayout" name="formLayout_5">
           <item row="0" column="0">
            <widget class="QLabel" name="label_2">
             <property name="toolTip">
              <string>Color of the light. Best to chose a higher than one value (the higher, the brighter)</string>
             </property>
             <property name="text">
              <string>Color</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QLineEdit" name="lineLightR">
               <property name="placeholderText">
                <string>Red part of the color (&gt;0.0)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineLightG">
               <property name="placeholderText">
                <string>Green part of the color (&gt;0.0)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineLightB">
               <property name="placeholderText">
                <string>Blue part of the color (&gt;0.0)</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QPushButton" name="buttonAddObject">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Add Sphere</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Label</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="lineSphereLabel">
          <property name="toolTip">
           <string>Label of the sphere, must be unique</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Radius</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="lineRadius">
          <property name="toolTip">
           <string>Radius of the sphere</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="placeholderText">
           <string>Radius of the sphere</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Plane</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
